packages/react/src/products/hooks/useProductsList.ts

```diff
@@ -0,0 +1,126 @@
+/**
+ * Hook to provide all kinds of data for the business logic attached to a product list.
+ *
+ * @module useProductsList
+ * @category Products
+ * @subcategory Hooks
+ */
+import {
+  fetchListing,
+  fetchSet,
+  getProductsListError,
+  getProductsListProducts,
+  getProductsListResult,
+  isProductsListLoading,
+} from '@farfetch/blackout-redux/products';
+import { generateProductsListHash } from '@farfetch/blackout-redux/products/utils';
+import { useDispatch, useSelector } from 'react-redux';
+import { useEffect } from 'react';
+import type {
+  ListingQuery,
+  SetQuery,
+} from '@farfetch/blackout-client/products/types';
+import type {
+  ProductsListTypes,
+  UseProductsList,
+  UseProductsListParams,
+} from './types/useProductsList';
+
+export const PRODUCTS_LIST_TYPES: ProductsListTypes = {
+  LISTING: 'listing',
+  SET: 'set',
+};
+
+/**
+ * Hook to handle listing logic.
+ *
+ * @param {object} props - Object containing the necessary info to use inside the hook.
+ * @param {string} [props.slug] - The slug of the given product list.
+ * @param {Object} [props.query] - The query parameters of the given product list.
+ * @param {string} [props.type='listing'] - Listing type ('listing' or 'set').
+ * @param {boolean} [props.useCache=true] - To use cache when fetching a product list.
+ * @param {boolean} [props.setProductsListHash=true] - Wether to set the reducer's hash on the store when requesting a product list.
+ *
+ * @returns {Object} All the info needed to get listing results and information.
+ */
+const useProductsList = ({
+  query,
+  setProductsListHash = true,
+  slug,
+  type = PRODUCTS_LIST_TYPES.LISTING,
+  useCache = true,
+}: UseProductsListParams): UseProductsList => {
```

Comment:
If you accept my previous suggestion we simplify this to:

```suggestion
const useProductsList: useProductsList = ({
  query,
  setProductsListHash = true,
  slug,
  type = PRODUCTS_LIST_TYPES.LISTING,
  useCache = true,
}) => {
```

Reply:
I know eheh thanks! 😁